### PR TITLE
Add a bound to the reaper while loop

### PR DESCRIPTION
**Pull request by EricWittmann**
Slight change to the tenant data reaper so that its control loop does not have a potentially unbounded while loop.

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/mt/TenantReaper.java [hunk 1]
```diff
@@ -111,12 +113,23 @@ void run() {
...
+    /**
+     * Query the tenant manager for the list of tenants in the "to-be-deleted" state.  Those tenants
+     * must be "reaped", which simply means we must delete all of their user data (artifacts, global
+     * rule configuration, etc).  This method is invoked by the scheduler and works by making an API
+     * call to the tenant manager and iterating through the results.
+     *
+     * Note that a single invocation of reap() will reap a maximum of MAX_TENANTS_PROCESSED.  If there
+     * are more tenants that need reaping, they will be processed the next time the schedule warrants it.
+     * This is a defensive approach to ensure that the while loop is always bounded.
+     */
```
Comment (EricWittmann) [inline app/src/main/java/io/apicurio/registry/mt/TenantReaper.java:125]: Javadoc!  :)

File: storage/kafkasql/src/main/java/io/apicurio/registry/storage/impl/kafkasql/values/ActionType.java [hunk 2]
```diff
@@ -32,11 +32,7 @@
     DELETE(3),
     CLEAR(4),
     IMPORT(5),
-    RESET(6),
-    /**
-     * Deletes ALL user (tenant) data. Does not delete global data, such as log configuration.
-     */
-    DELETE_ALL_USER_DATA(7);
+    RESET(6);
```
Comment (EricWittmann) [inline storage/kafkasql/src/main/java/io/apicurio/registry/storage/impl/kafkasql/values/ActionType.java:35]: What do you think about this?  I wasn't sure we needed the `DELETE_ALL_USER_DATA` enum when we already have `DELETE` which, when combined with the `GlobalAction` type seemed sufficient.

Thoughts?

Comment (jsenko) [reply storage/kafkasql/src/main/java/io/apicurio/registry/storage/impl/kafkasql/values/ActionType.java:35]: No strong opinion, I think the original name is more explicit about what it's doing. Sometimes words can get too overloaded.

File: app/src/main/java/io/apicurio/registry/mt/TenantReaper.java [hunk 3]
```diff
@@ -139,7 +152,8 @@ void reap() {
                     // Just ignore, will retry on next cycle
                 }
             }
-        } while (!page.isEmpty());
+            tenantsProcessed += tenants.getItems().size();
```
Comment (jsenko) [inline app/src/main/java/io/apicurio/registry/mt/TenantReaper.java:155]: Use `page.size()`

**Review by jsenko — APPROVED** [review]
Looks good. Would it be worth it to make `MAX_TENANTS_PROCESSED` configurable? On the other hand we can always run the reaper more often.
Could you also delete `void setNext(Instant next)` method, which I forgot to do after it was no longer used.

File: storage/kafkasql/src/main/java/io/apicurio/registry/storage/impl/kafkasql/values/ActionType.java (hunk 2 above)
Comment (EricWittmann) [reply storage/kafkasql/src/main/java/io/apicurio/registry/storage/impl/kafkasql/values/ActionType.java:35]: Yeah I think I'll revert this change.

**Comment by EricWittmann** [comment]
@jsenko I have reverted the kafkasql changes and made suggested changes.  If tests pass and you're happy, please merge this at your convenience.

**Pull request merged**